src/screens/SettingsSheet/components/Backups/BackUpMenuButton.tsx

```diff
@@ -0,0 +1,106 @@
+import { useCreateBackupStateType } from '@/components/backup/useCreateBackup';
+import { useTheme } from '@/theme';
+import React, { useState, useMemo, useEffect } from 'react';
+import * as i18n from '@/languages';
+import MenuItem from '../MenuItem';
+import Spinner from '@/components/Spinner';
+import { FloatingEmojis } from '@/components/floating-emojis';
+import { useDimensions } from '@/hooks';
+
+export const BackUpMenuItem = ({
+  icon = '􀊯',
+  loading,
+  onPress,
+}: {
+  icon?: string;
+  loading: useCreateBackupStateType;
+  title: string;
+  onPress: () => void;
+}) => {
+  const { colors } = useTheme();
+  const { width: deviceWidth } = useDimensions();
+  const [emojiTrigger, setEmojiTrigger] = useState<null | (() => void)>(null);
+
+  useEffect(() => {
+    console.log({ loading });
+    if (loading === 'success') {
+      for (let i = 0; i < 20; i++) {
+        setTimeout(() => {
+          emojiTrigger?.();
+        }, 100 * i);
+      }
+    }
+  }, [emojiTrigger, loading]);
+
+  const accentColor = useMemo(() => {
+    switch (loading) {
+      case 'success':
+        return colors.green;
+      case 'error':
+        return colors.red;
+      default:
+        return undefined;
+    }
+  }, [colors, loading]);
+
+  const titleText = useMemo(() => {
+    switch (loading) {
+      case 'loading':
+        return i18n.t(i18n.l.back_up.cloud.backing_up);
+      case 'success':
+        return i18n.t(i18n.l.back_up.cloud.backup_success);
+      case 'error':
+        return i18n.t(i18n.l.back_up.cloud.backup_failed);
+      default:
+        return i18n.t(i18n.l.back_up.cloud.enable_cloud_backups);
```

Comment:
I think it would make more sense if we were to use the incoming title prop as the default here. What do you think?

Reply:
yeah its supposed to, nice catch